File: src/cli/help/exec.ts
import type { CommandHelp } from "../help.ts";

export const execHelp: CommandHelp = {
  name: "exec",
  description: "Execute a command in a worktree directory",
  usage: "phantom exec [options] <worktree-name> <command> [args...]",
  options: [
    {
      name: "--fzf",
      type: "boolean",
      description: "Use fzf for interactive worktree selection",
    },
    {
      name: "--tmux, -t",
      type: "boolean",
      description: "Execute command in new tmux window",
    },
    {
      name: "--tmux-vertical, --tmux-v",
      type: "boolean",
      description: "Execute command in vertical split pane (tmux)",
    },
    {
      name: "--tmux-horizontal, --tmux-h",
      type: "boolean",
      description: "Execute command in horizontal split pane (tmux)",
    },
    {
      name: "--kitty, -k",
      type: "boolean",
      description: "Execute command in new kitty tab",
    },
    {
      name: "--kitty-vertical, --kitty-v",
      type: "boolean",
      description: "Execute command in vertical split (kitty)",
    },
    {
      name: "--kitty-horizontal, --kitty-h",
      type: "boolean",
      description: "Execute command in horizontal split (kitty)",
    },
  ],
  examples: [
    {
      description: "Run npm test in a worktree",
      command: "phantom exec feature-auth npm test",
    },
    {
      description: "Check git status in a worktree",
      command: "phantom exec bugfix-123 git status",
    },
    {
      description: "Run a complex command with arguments",
      command: "phantom exec staging npm run build -- --production",
    },
    {
      description: "Execute with interactive selection",
      command: "phantom exec --fzf npm run dev",
    },
    {
      description: "Run dev server in new tmux window",
      command: "phantom exec --tmux feature-auth npm run dev",
    },
    {
      description: "Run tests in vertical split pane",
      command: "phantom exec --tmux-v feature-auth npm test",
    },
    {
      description: "Run dev server in new kitty tab",
      command: "phantom exec --kitty feature-auth npm run dev",
    },
    {
      description: "Run tests in vertical kitty split",
      command: "phantom exec --kitty-v feature-auth npm test",
    },
    {
      description: "Interactive selection with tmux",
      command: "phantom exec --fzf --tmux npm run dev",
    },
  ],
  notes: [
    "The command is executed with the worktree directory as the working directory",
    "All arguments after the worktree name are passed to the command",
    "The exit code of the executed command is preserved",
    "With --fzf, select the worktree interactively before executing the command",
    "Tmux options require being inside a tmux session",
    "Kitty options require being inside a kitty terminal",
    "Tmux and kitty options cannot be used together",
  ],
};

File: rust/src/cli/commands/exec.rs
use clap::Args;

#[derive(Args, Debug)]
pub struct ExecArgs {
    /// Name of the worktree (can be omitted with --fzf)
    pub name: Option<String>,

    /// Command and arguments to execute
    #[arg(trailing_var_arg = true, allow_hyphen_values = true, required = true)]
    pub command: Vec<String>,

    /// Select worktree interactively with fzf
    #[arg(long)]
    pub fzf: bool,

    /// Execute in a new tmux window
    #[arg(short = 't', long)]
    pub tmux: bool,

    /// Execute in a vertical tmux pane
    #[arg(long = "tmux-vertical", conflicts_with = "tmux")]
    pub tmux_vertical: bool,

    /// Execute in a vertical tmux pane (shorthand)
    #[arg(long = "tmux-v", conflicts_with_all = &["tmux", "tmux_vertical"])]
    pub tmux_v: bool,

    /// Execute in a horizontal tmux pane
    #[arg(long = "tmux-horizontal", conflicts_with_all = &["tmux", "tmux_vertical", "tmux_v"])]
    pub tmux_horizontal: bool,

    /// Execute in a horizontal tmux pane (shorthand)
    #[arg(long = "tmux-h", conflicts_with_all = &["tmux", "tmux_vertical", "tmux_v", "tmux_horizontal"])]
    pub tmux_h: bool,

    /// Execute in a new kitty tab
    #[arg(short = 'k', long, conflicts_with_all = &["tmux", "tmux_vertical", "tmux_v", "tmux_horizontal", "tmux_h"])]
    pub kitty: bool,

    /// Execute in a vertical kitty split
    #[arg(long = "kitty-vertical", conflicts_with_all = &["tmux", "tmux_vertical", "tmux_v", "tmux_horizontal", "tmux_h", "kitty"])]
    pub kitty_vertical: bool,

    /// Execute in a vertical kitty split (shorthand)
    #[arg(long = "kitty-v", conflicts_with_all = &["tmux", "tmux_vertical", "tmux_v", "tmux_horizontal", "tmux_h", "kitty", "kitty_vertical"])]
    pub kitty_v: bool,

    /// Execute in a horizontal kitty split
    #[arg(long = "kitty-horizontal", conflicts_with_all = &["tmux", "tmux_vertical", "tmux_v", "tmux_horizontal", "tmux_h", "kitty", "kitty_vertical", "kitty_v"])]
    pub kitty_horizontal: bool,

    /// Execute in a horizontal kitty split (shorthand)
    #[arg(long = "kitty-h", conflicts_with_all = &["tmux", "tmux_vertical", "tmux_v", "tmux_horizontal", "tmux_h", "kitty", "kitty_vertical", "kitty_v", "kitty_horizontal"])]
    pub kitty_h: bool,
}
